src/lib.rs

```diff
@@ -667,6 +691,47 @@ mod tests {
         let mut ledger = FULL_LEDGER.clone();
         assert_eq!(ledger.records(), &[FULL]);
 
+        println!("{:#?}", &ledger);
+        println!("Unmaps:");
+        trace_regions(&unmaps);
+
+        for region in unmaps {
+            let addr = region.start;
+            let length = region.end - region.start;
+            ledger.unmap(addr, length).unwrap();
+        }
+
+        println!("Result:");
+        println!("{:#?}", &ledger);
+        trace_assert_records_eq(ledger.records(), &expected);
+    }
+
+    #[rstest::rstest]
+    #[case(&[(0x8, 0x10)], &[(0x0, 0x8, R)])] // start
+    #[case(&[(0x0, 0x8)], &[(0x8, 0x10, W)])] // end
+    #[case(&[(0x8, 0x9)], &[(0x0, 0x8, R), (0x9, 0x10, W)])] // split
+    #[case(&[(0x8, 0x9), (0x8, 0x10)], &[(0x0, 0x8, R)])] // split and end
+    #[case(&[(0x7, 0x9)], &[(0x0, 0x7, R), (0x9, 0x10, W)])] // split
+    fn unmap_mixed(#[case] unmaps: &[(usize, usize)], #[case] expected: &[(usize, usize, Access)]) {
+        let unmaps = unmaps
+            .iter()
+            .cloned()
+            .map(|record| Region::new(Address::new(record.0 << 12), Address::new(record.1 << 12)))
+            .collect::<Vec<_>>();
+
+        let expected = expected
+            .iter()
+            .cloned()
+            .map(|record| Record {
+                region: Region::new(Address::new(record.0 << 12), Address::new(record.1 << 12)),
+                access: record.2,
+            })
+            .collect::<Vec<_>>();
```

Comment:
The two statements above look like boilerplate shared among all the tests, do you think they should be split out into `to_regions` and `to_records` functions?

Reply:
done